controllers/vspheremachine_controller.go

```diff
@@ -260,11 +264,13 @@ func (r *machineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 }
 
 func (r *machineReconciler) reconcileDelete(ctx context.Context, machineCtx capvcontext.MachineContext) (reconcile.Result, error) {
-	log := ctrl.LoggerFrom(ctx)
+	log := ctrl.LoggerFrom(ctx).WithValues("VSphereMachine", klog.KRef(machineCtx.GetVSphereMachine().GetNamespace(), machineCtx.GetVSphereMachine().GetName()))
```

Comment:
Shouldn't this already be in the logger? via controller-runtime?

Reply:
How do we find out whether the objects should already be in logger by controller-runtime?

Reply:
I think this gets done here: https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/internal/controller/controller.go#L306

Which propably uses: https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/controller/controller.go#L118-L125

Or from this PR's e2e job capv logs:

```
I0929 06:59:53.813727       1 vspheremachine_controller.go:183] "Starting Reconcile" controller="vspheremachine-controller" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="VSphereMachine" VSphereMachine="quick-start-z9hm66/quick-start-boxouv-md-0-infra-7n7vm-g785p" namespace="quick-start-z9hm66" name="quick-start-boxouv-md-0-infra-7n7vm-g785p" reconcileID=a7d09514-d36a-444c-8745-9d7866523b29
```

`VSphereMachine="quick-start-z9hm66/quick-start-boxouv-md-0-infra-7n7vm-g785p" ` duplicates the information which is injected from controller-runtime already. SO we don't need to add it.

Reply:
[source](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-vsphere/2395/pull-cluster-api-provider-vsphere-e2e-main/1707649667753316352/artifacts/clusters/bootstrap/logs/capv-system/capv-controller-manager/capv-controller-manager-56d5dbbc5b-2dftl/manager.log) for the above log line

Reply:
Thankyou 🙂 this helps